net-dns/yaddns/patches/yaddns-1.1.patchset

```diff
@@ -0,0 +1,243 @@
+From ee1290f65761f521bde5fc574a0690e9e7489891 Mon Sep 17 00:00:00 2001
+From: sfanxiang <sfanxiang@gmail.com>
+Date: Mon, 8 Jan 2018 10:10:48 +0000
+Subject: port to Haiku
+
+
+diff --git a/configure.in b/configure.in
+index d8c6668..fd6b42d 100644
+--- a/configure.in
++++ b/configure.in
+@@ -9,6 +9,7 @@ test ".$CFLAGS" = "." && CFLAGS=" "
+ 
+ # programs needed to build/install
+ AC_PROG_CC
++AC_PROG_CXX
+ AC_PROG_INSTALL
+ AC_PROG_RANLIB
+ 
+@@ -66,6 +67,8 @@ if test "x$enable_log_color" = "xyes"; then
+    CFLAGS="$CFLAGS -DENABLE_LOG_COLOR"
+ fi
+ 
++CXXFLAGS=$CFLAGS
++
+ # the generated files
+ AC_CONFIG_FILES([
+ Makefile
+diff --git a/src/Makefile.am b/src/Makefile.am
+index 750ae3a..36e65e9 100644
+--- a/src/Makefile.am
++++ b/src/Makefile.am
+@@ -1,3 +1,5 @@
++AM_LDFLAGS = -lbnetapi -lnetwork
++
+ SUBDIRS = services
+ 
+ bin_PROGRAMS = yaddns
+@@ -5,6 +7,7 @@ bin_PROGRAMS = yaddns
+ yaddns_SOURCES = yaddns.c yaddns.h \
+ 	config.c config.h \
+ 	account.c account.h \
++	ifaddr.cc ifaddr.h \
+ 	request.c request.h \
+ 	log.c log.h \
+ 	util.c util.h \
+diff --git a/src/config.c b/src/config.c
+index 8f7b324..66c2a52 100644
+--- a/src/config.c
++++ b/src/config.c
+@@ -7,13 +7,15 @@
+ #include <getopt.h>
+ #include <limits.h>
+ 
++#include <FindDirectory.h>
++#include <fs_info.h>
++
+ #include "config.h"
+ #include "log.h"
+ #include "service.h"
+ #include "services.h"
+ #include "util.h"
+ 
+-#define CFG_DEFAULT_FILENAME "/etc/yaddns.conf"
+ #define CFG_DEFAULT_WANIFNAME "ppp0"
+ 
+ /*
+@@ -228,7 +230,17 @@ int config_parse(struct cfg *cfg, int argc, char **argv)
+ 
+         if(!cfgstr_is_set(&(cfg->cfgfile)))
+         {
+-                cfgstr_set(&(cfg->cfgfile), CFG_DEFAULT_FILENAME);
++                char filename[] = "/yaddns/yaddns.conf";
++
++                dev_t vol = dev_for_path("/boot");
++                static char buf[B_PATH_NAME_LENGTH + B_FILE_NAME_LENGTH
++                    + sizeof(filename)];
++
++                find_directory(B_SYSTEM_SETTINGS_DIRECTORY,
++                    vol, 0, buf, sizeof(buf));
```

Comment:
`find_directory` is not necessary when dealing with system paths. You could just swap `CFG_DEFAULT_FILENAME` value with `$settingsDir` and that would be just fine

Reply:
OK, I see what you mean. We could add `-DCFG_DEFAULT_FILENAME=\""$settingsDir"/yaddns/yaddns.conf\"` to the CFLAGS.
Well, there is a small downside with this approach: if/when the system paths change, the program would need to be rebuilt.
So, since the patch to use find_directory is already working, we can probably keep it. This would reduce the cost of changing the system paths, if we ever need to do that.

Reply:
If the system paths ever changes, most ports (if not all) would have to be rebuilt. Given that we've already hard-coded system paths in the form of `configure` flags (`--sysconfdir`) for potentially every ports that uses `runConfigure`, I don't really see this as a problem.

While the patch is working, I would like to reduce the maintainance cost by reducing as much changes to the original source as possible.

Reply:
Again, please don't increase the amount of hardcoded paths. Using find_directory at runtime is always the right thing to do.

Reply:
Also, sysconfDir uses a package link dir, which is relocatable by the packagefs. So it should not be pointing at a fixed system directory. Did you see it configured otherwise?

Reply:
@mmlr 
Sorry, I thought sysconfdir was `$settingsDir`.

So I believe we should instead change `CFG_DEFAULT_FILENAME` to `$portPackageLinksDir/.settings`

`find_directory` can fail. And that means you would have to dealt with the fact that you would have no directory to use or to fallback to some sane defaults (`$portPackageLinksDir/.settings`).

Since this directory is relocatable by packagefs, I don't see any reason against using said path instead of more code.

---

P/s: Probably `$portPackageLinksDir/.settings` needs own variable.

Also, I would like to know the differences between `$settingsDir` and `$portPackageLinksDir/.settings` if possible, as the docs are scarce about this

Reply:
HaikuPorter defines:
* `portPackageLinksDir` as `/packages/$portVersionedName-$REVISION`
* `settingsDir` as `/packages/$portVersionedName-$REVISION/.self/settings`

So the difference between `$portPackageLinksDir/.settings` and `$settingsDir` is just that the former ends with `/.settings` while the latter ends with `/.self/settings`.

BTW, everytime I need to answer a question on this topic, I go to the work directory of a package I'm building and view its wrapper-script.

Reply:
I know. I was asking for functional differences, which I figured out myself after a minute or so :laughing:

Reply:
[`find_directory`][0] could fail, so please handle that case if possible

[0]: https://www.haiku-os.org/docs/api/FindDirectory_8h.html#af95c55d472448ab7af9f32c33aeea71e